doc/guides/build/kconfig/setting.rst

```diff
@@ -167,6 +167,22 @@ The application configuration can come from the sources below. By default,
 6. Otherwise, :file:`prj.conf` is used if it exists in the application
    directory
 
+Per default all configuration files will be taken from the application's
+configuration directory except for files with an absolute path that are
+given with the ``CONF_FILE`` argument.
+
+The application's configuration directory will be defined as:
```

Comment:
APPLICATION_CONFIG_DIR will be set by one of the sources below with the highest priority listed first.